lib/modules/manager/bazel-module/fragments.ts

```diff
@@ -0,0 +1,116 @@
+import { z } from 'zod';
+import * as starlark from './starlark';
+
+export const StringFragmentSchema = z.object({
+  type: z.literal('string'),
+  value: z.string(),
+  isComplete: z.literal(true),
+});
+export const BooleanFragmentSchema = z.object({
+  type: z.literal('boolean'),
+  value: z.boolean(),
+  isComplete: z.literal(true),
+});
+const PrimitiveFragmentsSchema = z.discriminatedUnion('type', [
+  StringFragmentSchema,
+  BooleanFragmentSchema,
+]);
+export const ArrayFragmentSchema = z.object({
+  type: z.literal('array'),
+  items: PrimitiveFragmentsSchema.array(),
```

Comment:
For array and record schemas, single invalid element causes the whole array or record to be rejected. To filter out invalid elements without giving up on the whole container, please take a look at `LooseArray` and `LooseRecord` in `schema-utils.ts`.